prime-router/docs/design/design/deduplication/deduplication.md

```diff
@@ -0,0 +1,197 @@
+
+Technical Considerations:
+- This should be flexible enough that different senders, message types, and topics can implement different key fields.
+- The ordering of fields within a FHIR Bundle are **not** guaranteed. Before converting to a string to be hashed, the [populated key fields](#key-fhir-elements) should be extracted from the bundle and put in a static order (alphabetical etc.).
+- Sender id/name needs to be incorporated into pre-hashed string. (Dependent upon sql efficiency investigation noted below).
+- Investigation TODO (_To be removed and appropriate sections updated before merge_): Investigate the time efficiency of the following options:
+  - Adding a sender id column to item_lineage table (and using this to narrow the SQL query).
+  - Adding the sender id (or name) into the string to be hashed and allowing the column index to be the main SQL query parameter.
+  - Performing a SQL join to a table with the sender id to narrow the query. 
+  - Notes:
+    - (Date range would also be applied to the SQL query in all scenarios.) 
+    - (Comment from Arnej: "Evaluate the performance, quality, and security trade-off of including the sender name in the hash vs querying it.")
+    - (Comment from Michael: "For this to be performative you would need to be sure that this index can fit in memory of postgresql instead of having to go into disk.")
+
+##### Item Hash Comparison
+Hash comparison will be skipped if deduplication is disabled on that sender's settings. The generated item hash shall first be compared with items in the same report. (Note: At this part of the Convert Step, parallelization is not a blocker as the original items are [still within scope](https://github.com/CDCgov/prime-reportstream/blob/15648395efc2b60322d931bf88e0c2c5b6cc0371/prime-router/src/main/kotlin/fhirengine/engine/FHIRConverter.kt#L510)). The item hash will then be compared to existing hashes from the item_lineage table. There is an [existing SQL Query](https://github.com/CDCgov/prime-reportstream/blob/9ec0a59c73d7dad9a319cd321baf9efd71ceab46/prime-router/src/main/kotlin/azure/DatabaseAccess.kt#L166-L183) which performs the hash comparison through a database query. This will need to be enhanced or recreated as the original query only searches within the last 7 days and does not take sender into account. (_This last part will be unnecessary if the sender is incorporated into the hash._) 
```

Comment:
Define the setting name

Reply:
Somehow I previously missed that we should be able to entirely repurpose the covid pipeline sender setting for this. I've updated this document to reflect that, as well as a couple other docs that made reference to it in the context of the UP. https://github.com/CDCgov/prime-reportstream/pull/17143/commits/798d27495092457188697aa05e99a021d1bd3a9b